Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/PowerTransmissions/Axle.cs

```diff
@@ -707,117 +815,134 @@ public void Save(BinaryWriter outf)
         /// <summary>
         /// Compute variation in axle dynamics. Calculates axle speed, axle angular position and in/out forces.
         /// </summary>
-        public (float, float, float, float) GetAxleMotionVariation(float axleSpeedMpS)
+        public (double, double, double, double) GetAxleMotionVariation(double axleSpeedMpS, double elapsedClockSeconds)
         {
-            float axleOutForceN = AxleWeightN * SlipCharacteristics(axleSpeedMpS - TrainSpeedMpS, TrainSpeedMpS, AdhesionK, AdhesionLimit);
+            double slipSpeedMpS = axleSpeedMpS - TrainSpeedMpS;
+            double axleOutForceN = 0;
 
-            float axleInForceN = 0;
+            if (UsePolachAdhesion)
+            {
+                axleOutForceN = Math.Sign(slipSpeedMpS) * AxleWeightN * SlipCharacteristicsPolach(slipSpeedMpS);
+            }
+            else
+            {
+                axleOutForceN = AxleWeightN * SlipCharacteristicsPacha((float)axleSpeedMpS - TrainSpeedMpS, TrainSpeedMpS, AdhesionK, AdhesionLimit);
+            }
```

Comment:
I would prefer if the differences between models are kept in SlipCharacteristics(), so the rest of the functions is platform-agnostic. I offer to do that in a further PR if you prefer. Only a couple of lines have to be changed.